data/src/main/java/co/nimblehq/compose/crypto/data/repository/GlobalRepositoryImpl.kt

```diff
@@ -0,0 +1,53 @@
+package co.nimblehq.compose.crypto.data.repository
+
+import android.content.Context
+import android.net.*
+import co.nimblehq.compose.crypto.domain.repository.GlobalRepository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+
+class GlobalRepositoryImpl(
+    context: Context
+) : GlobalRepository {
+
+    private var connectionStatus: MutableStateFlow<Boolean?> = MutableStateFlow(null)
```

Comment:
Perhaps we could call this `isNetworkConnected` or similar? `State` usually implies more than just true or false IMO, what do you think? 🙏🏻